e2e/TROUBLESHOOTING.md

```diff
@@ -5,10 +5,88 @@ The first step in determining the exact failure is to try and reproduce the test
+
+To get the IP address of the VMs, please go to the Jenkins'm BlueOcean UI of the job you manually triggered with `DEVELOPER_MODE=true`, and look up the **Deploy Test Infra** stage. Under its steps, you will see different Ansible executions to provision the VM. Please open any of them and look for any log entry containing an IP address:
+
+```shell
+[2022-04-21T05:20:27.133Z] TASK [Gathering Facts] *********************************************************
+[2022-04-21T05:20:29.541Z] ok: [3.144.74.102]
+```
+
+The IP address of that VM is `3.144.74.102`.
+
+For the agent VMs it's exactly the same, but looking up the **parallel stages** right after the Test Infra. Again, look for any Ansible task containing an IP address, as shown above.
+
+Once you have both IP addresses, one for the stack and one for the agent in the OS/Arch you are interested in, please open two terminals, one for each. Then SSH into the machines using: 1) the public SSH key you have in your Github account, and 2) the right user for the machine, as described [here](https://github.com/elastic/e2e-testing/blob/4517dfa134844f720139d6bab3955cc8d9c6685c/.ci/.e2e-platforms.yaml#L2-L42).
+
+An example of how to SSH in the machine, having multiple SSH keys under the `.ssh` directory, and connecting with the `admin` user because it's a Debian machine:
+
+```shell
+ssh -i ~/.ssh/id_rsa_elastic.pub -vvvv admin@18.188.242.30
+```
+
+#### Running the tests for the Elastic Agent
+Once you have SSH'ed into the right agent VM (ex. CentOS 8 ARM), and checked that the stack is in a valid state (the API token of the Fleet Server has not expired, otherwise you must recreate the entire stack), you can run the tests with a simple command, but you must first set the environment so that the agent is able to connect to the remote stack, which lives in another VM. Just source the `.env` file that the test framework created:
+
+```shell
+# log in as root
+sudo su -
+# move to the project directory
+cd /home/${USER}/e2e-testing
+# load the environment
+source .env
+# verify variables
+env
+```
+
+The env should contain those variables needed for enroling the agent, such as `ELASTICSEARCH_URL` and `FLEET_SERVER_URL`, among others.
+
+Now you can run the tests, specifying the tags you are interested. Please use the tags in the feature files, where the test framework defines one at the top level, for running an entire feature file, and per scenario, so that it's possible to tell the test runner to run one or multiple scenarios. More about Cucumber tags in [here](https://github.com/cucumber/godog#tags).
+
```

Comment:
[Suggestion]
I'd suggest to explain better what are the tags and how to find them. [The docs ](https://github.com/cucumber/godog#tags) explain only how to use them, not how to define them. I had a hard time to find where the tags were. So this is my suggestion:

```suggestion
Now you can run the tests, specifying the tags you are interested. Please use the tags in the feature files.
The test framework defines a tag at the top level, for running an entire feature file, and one per scenario, so that it's possible to tell the test runner to run one or multiple scenarios.
The tags are the `@tag-name` [found at the very top of the `.feature` files](https://github.com/elastic/e2e-testing/blob/main/e2e/_suites/fleet/features/fleet_mode_agent.feature#L1) or [above each individual test](https://github.com/elastic/e2e-testing/blob/main/e2e/_suites/fleet/features/system_integration.feature#L11).
More about Cucumber tags in [here](https://github.com/cucumber/godog#tags).

```